[quant] Correct name for quantized LSTM

z-a-f opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #64834
* #64833

Once quantized, the quantizable LSTM is still showing up as "QuantizableLSTM". It would be more correct to change it to "QuantizedLSTM". Note: This works for both converting with custom module and converting without the custom module (QUantizableLSTM supports both ways).

Differential Revision: [D30872072](https://our.internmc.facebook.com/intern/diff/D30872072/)

Commit:
[quant] Correct name for quantized LSTM

Once quantized, the quantizable LSTM is still showing up as "QuantizableLSTM". It would be more correct to change it to "QuantizedLSTM". Note: This works for both converting with custom module and converting without the custom module (QUantizableLSTM supports both ways).

Differential Revision: [D30872072](https://our.internmc.facebook.com/intern/diff/D30872072/)

[ghstack-poisoned]

Commit:
Update on "[quant] Correct name for quantized LSTM"


Once quantized, the quantizable LSTM is still showing up as "QuantizableLSTM". It would be more correct to change it to "QuantizedLSTM". Note: This works for both converting with custom module and converting without the custom module (QUantizableLSTM supports both ways).

Differential Revision: [D30872072](https://our.internmc.facebook.com/intern/diff/D30872072/)

[ghstack-poisoned]

vkuzo reviewed (changes requested):
I see this was approved, should we discuss first if this if this PR is a net positive?  It's a bit confusing if a module's name is dynamic, it's not how other things in PyTorch work.  Can we add some motivation on why we need this and why it's worth diverging from normal?

Commit:
Update on "[quant] Correct name for quantized LSTM"


Once quantized, the quantizable LSTM is still showing up as "QuantizableLSTM". It would be more correct to change it to "QuantizedLSTM". Note: This works for both converting with custom module and converting without the custom module (QUantizableLSTM supports both ways).

Differential Revision: [D30872072](https://our.internmc.facebook.com/intern/diff/D30872072/)

[ghstack-poisoned]

vkuzo reviewed (approved):
looks like I myself asked for this some time ago and then am unhappy with it, my fault.  Accepting to unblock.

Commit:
Update on "[quant] Correct name for quantized LSTM"


Once quantized, the quantizable LSTM is still showing up as "QuantizableLSTM". It would be more correct to change it to "QuantizedLSTM". Note: This works for both converting with custom module and converting without the custom module (QUantizableLSTM supports both ways).

Differential Revision: [D30872072](https://our.internmc.facebook.com/intern/diff/D30872072/)

[ghstack-poisoned]
